COM-795 Update function patientHIVFinalTestResultIsWithinReportingPeriod

lauravignoli opened the pull request:
Use HIV Test Date in HIV Section to check that Final result is within the reporting period.

lauravignoli committed:
COM-795 Update function patientHIVFinalTestResultIsWithinReportingPeriod

Use HIV Test Date in HIV Section to check that Final result is within the reporting period.

tmvumbi2 commented on metadata/reportssql/testing_report_functions.sql:
```diff
@@ -1269,6 +1269,15 @@ BEGIN
+    DECLARE uuidHIVTestSection VARCHAR(38) DEFAULT "b70dfca0-db21-4533-8c08-4626ff0de265";
+    DECLARE withinReportingPeriod TINYINT(1) DEFAULT 0;
+
+    SET withinReportingPeriod = getObsDatetimeValueInSection(p_patientId, uuidHIVTestDate, uuidHIVTestSection) BETWEEN p_startDate AND p_endDate;
```
@lauravignoli  the function `getObsDatetimeValueInSection` returns `NULL` when the date doesn't exist; and the result of `NULL BETWEEN value1 AND value` is actually NULL (i.e. not 0 as I also intuitively expected it). The condition in the next line therefore also returns NULL, causing `RETURN 0` not to be called

tmvumbi2 commented on metadata/reportssql/testing_report_functions.sql:
```diff
@@ -1269,6 +1269,15 @@ BEGIN
+    DECLARE withinReportingPeriod TINYINT(1) DEFAULT 0;
+
+    SET withinReportingPeriod = getObsDatetimeValueInSection(p_patientId, uuidHIVTestDate, uuidHIVTestSection) BETWEEN p_startDate AND p_endDate;
+    IF NOT withinReportingPeriod THEN
```
@lauravignoli a quick fix could be to use `IF withinReportingPeriod IS NULL THEN` here

lauravignoli committed:
COM-795 Fix condition

The function getObsDatetimeValueInSection returns NULL when the date doesn't exist; and the result of NULL BETWEEN value1 AND value is actually NULL (i.e. not 0 as I also intuitively expected it). The condition in the next line therefore also returns NULL, causing RETURN 0 not to be called.

Fixed it by checking if it's null instead of false.
